src/components/moderation/moderation-common.ts

```diff
@@ -62,50 +62,76 @@ export const joke_responses = [
     "Didn't work. Maybe a skill issue?",
 ];
 
-// returns duration in ms
-function parse_unit(u: string) {
+function millis_of_time_unit(u: string) {
     let factor = 1;
     switch (u) {
         case "y":
+        case "year":
+        case "years":
             factor *= 365; // 365 days, fallthrough
         case "d":
+        case "day":
+        case "days":
             factor *= 24; // 24 hours, fallthrough
         case "h":
+        case "hr":
+        case "hour":
+        case "hours":
             factor *= 60; // 60 minutes, fallthrough
         case "m":
+        case "min":
+        case "mins":
+        case "minute":
+        case "minutes":
             factor *= 60; // 60 seconds, fallthrough
         case "s":
+        case "sec":
+        case "secs":
+        case "seconds":
             factor *= 1000; // 1000 ms
             break;
         // Weeks and months can't be folded into the above as nicely
         case "w":
-            factor *= 7 * parse_unit("d");
+        case "week":
+        case "weeks":
+            factor *= 7 * millis_of_time_unit("d")!;
             break;
         case "M":
         case "mo":
-            factor *= 30 * parse_unit("d");
+        case "month":
+        case "months":
+            factor *= 30 * millis_of_time_unit("d")!;
             break;
         default:
-            assert(false, "Unexpected unit");
+            return null;
     }
     return factor;
 }
 
-// Duration string to milliseconds
-export function parse_duration(duration: string | null) {
-    if (duration === null) {
-        return null;
-    }
+// Returns the corresponding duration in milliseconds,
+// or Infinity for permanent duration,
+// or null if parsing failed.
+export function parse_duration(duration: string) {
```

Comment:
I have some thoughts on the design here: On one hand it can make sense to treat a permanent duration as infinity but on the other hand other parts of the duration handling rely on `null` to represent a permanent duration, as you've found with having to convert from `string | null` and then back to `number | null`. My recommendation would be to just throw an error on an unknown unit, I think that will be simpler and require less back and forth and special casing. Introducing infinity here requires a lot of care to ensure the edge case is handled correctly which is especially error-prone when not reflected in the type system. In fairness, throwing an exception also isn't reflected in the type system but the failure mode of not catching is a bot reply saying something went wrong as opposed to some nasty error where an infinity makes it into the database or something.

Reply:
To be fair, I think using null to imply infinite duration is pretty janky, but it's too late to change the design now.

Converted to exceptions as requested.